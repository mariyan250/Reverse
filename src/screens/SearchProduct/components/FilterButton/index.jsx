// React and Style
import React from 'react';
import styles from './index.module.scss';

// Translation
import { useTranslation } from 'react-i18next';

// Components
import FilterDropdown from '../FilterDropdown';

function FilterButton({ isOpen, onClick, role }) {
  const { t } = useTranslation();

  return (
    <button
      className={`${styles.button} ${
        isOpen ? styles.toggle : ''
      } d-flex justify-content-center align-items-center position-relative box-shadow rounded p-2 border-0 app-bg-white`}
      onClick={onClick}
      role={role}
      aria-label="filter products"
    >
      <p>{t('search.filter_button')}</p>
      <i className="fas fa-angle-down ml-3" />
      {isOpen && <FilterDropdown />}
    </button>
  );
}

export default FilterButton;
