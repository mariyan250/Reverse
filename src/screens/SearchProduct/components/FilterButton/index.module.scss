@import '../../../../index.scss';

.button {
  font-size: to-rem(16px);
  width: to-rem(140px);
  box-shadow: $shadow-small;

  i {
    transition: $timing-normal;
    font-size: to-rem(24px);
  }

  &:focus {
    outline: none;
  }

  &:hover {
    cursor: pointer;
  }

  &.toggle {
    i {
      transform: rotate(180deg);
    }
  }
}
