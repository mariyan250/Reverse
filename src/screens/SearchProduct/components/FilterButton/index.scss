@import '../../../../index.scss';

.filter-button {
  button {
    font-size: to-rem(16px);
    width: to-rem(150px);

    i {
      transition: $normal;
      font-size: to-rem(24px);

      &.toggle {
        transform: rotate(180deg);
      }
    }

    &:focus {
      outline: none;
    }
  }

  &:hover {
    cursor: pointer;
  }
}
