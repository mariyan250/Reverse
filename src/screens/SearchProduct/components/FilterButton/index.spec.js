import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import FilterButton from './index';

describe('FilterButton', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { asFragment } = renderWithRouter(<FilterButton />, initialState);

      expect(asFragment(<FilterButton />)).toMatchSnapshot();
    });
  });
});
