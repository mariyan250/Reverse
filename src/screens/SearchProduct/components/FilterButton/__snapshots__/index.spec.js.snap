// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton should render the component 1`] = `
<div>
  <button
    aria-label="filter products"
    class="button  d-flex justify-content-center align-items-center position-relative rounded p-2 border-0 app-bg-white"
  >
    <p>
      search.filter_button
    </p>
    <i
      class="fas fa-angle-down ml-3"
    />
  </button>
</div>
`;
